crates/brainfuck_vm/src/bin/brainfuck_vm.rs

```diff
@@ -0,0 +1,58 @@
+// Adapted from rkdud007 brainfuck-zkvm https://github.com/rkdud007/brainfuck-zkvm/blob/main/src/main.rs
+
+use clap::{Parser, ValueHint};
+use std::{
+    fs,
+    io::{stdin, stdout},
+    path::PathBuf,
+};
+use tracing_subscriber::{fmt, EnvFilter};
+
+use brainfuck_vm::{compiler::Compiler, machine::Machine};
+
+#[derive(Parser)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    #[clap(value_parser, value_hint=ValueHint::FilePath)]
+    filename: PathBuf,
+    #[clap(long)]
+    trace: bool,
+    #[clap(long)]
+    pad_trace: bool,
+    #[clap(long)]
+    ram_size: Option<usize>,
+}
+
+fn main() {
+    let args = Args::parse();
+
+    // Constructs a subscriber whose severity level is filtered by `RUST_LOG`
+    fmt().with_env_filter(EnvFilter::from_default_env()).init();
+
+    let code = fs::read_to_string(&args.filename)
+        .unwrap_or_else(|_| panic!("Failed to read file"))
+        .replace(' ', "");
```

Comment:
is this required here? it looks like the compiler does it as well

Reply:
Indeed, and the compiler does it better (replace every whitespace rather than only the space character)